src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -453,7 +453,7 @@ class SchedulerActions(
       tasks.foreach {
         case task: Task =>
           if (task.launchedMesosId.isDefined) {
-            log.info("Killing {}", task.id)
+            log.info("Killing {}", task.instanceId)
```

Comment:
you're mapping over an iterable of instances here and match against `Task` – that won't work